java/vector/src/main/java/org/apache/arrow/vector/file/json/JsonFileReader.java

```diff
@@ -235,14 +237,16 @@ private void readVector(Field field, FieldVector vector) throws JsonParseExcepti
         nextFieldIs(vectorType.getName());
         readToken(START_ARRAY);
         ValueVector valueVector = (ValueVector) innerVector;
-        valueVector.allocateNew();
-        Mutator mutator = valueVector.getMutator();
 
         int innerVectorCount = vectorType.equals(OFFSET) ? count + 1 : count;
+        valueVector.setInitialCapacity(innerVectorCount);
```

Comment:
Why this?

Reply:
Most of the vectors do not use `setSafe` so reading beyond the default capacity would cause problems. Maybe for cases where the majority of values are null this is overdoing it, but I think that is less common.

Reply:
LGTM